### PR TITLE
Pull partner descriptions from JSON files

### PR DESCRIPTION
**Pull request by suecarmol**
## Description
Get the partner descriptions from a JSON file instead of the database and display them on the Django templates.

## Rationale
[//]: # (Why is this change required? What problem does it solve?)
Partner descriptions could not be translated to a user's language of choice. This change helps solve this.

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
[T278337](https://phabricator.wikimedia.org/T278337)

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
[//]: # (- Can this change be tested manually? How?)
1. Create a `partner_descriptions.json` file in a language other than English
2. Add a key-pair combination of your partner of choice
3. Change your settings to that language
4. Ensure that the partner description is in the chosen language

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)
**Partner description when the description exists in the user's language of choice**
Screen Shot 2021-04-05 at 17 35 47

**Partner description when the description does not exist in the user's language of choice**
Screen Shot 2021-04-05 at 17 36 00

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

**Comment by Samwalton9** [comment]
Just a quick thought - do we expect this to substantially impact the performance of the site, especially on pages like Browse where we're loading dozens of descriptions in one page load?

File: TWLight/resources/helpers.py [hunk 1]
```diff
@@ -83,3 +88,97 @@ def get_json_schema():
...
+
+    ----------
+    language_code: str
+        The language code the user has selected in their settings
+
+    Returns
+    -------
+    dict
+    """
+    home_dir = settings.TWLIGHT_HOME
```
Comment (jsnshrmn) [inline TWLight/resources/helpers.py:146]: super nitpicky, but we call this `twlight_home` in other places

**Review by jsnshrmn — APPROVED** [review]
This is looking really good to me. I think that it looks pretty much ready to go pending translatewiki having the i18n pipeline in place for this content.

**Comment by jsnshrmn** [comment]
> 
> 
> Just a quick thought - do we expect this to substantially impact the performance of the site, especially on pages like Browse where we're loading dozens of descriptions in one page load?

I'd expect it to be faster on the first page load after a clean deploy; in our current setup, those cached descriptions can get wiped (though they don't under normal circumstances). Regenerating them from db-side wikitext can take some time.
Otherwise I'd expect this to be fairly similar since both are ultimately serving prerendered html from disk.

**Review by jsnshrmn — COMMENTED** [review]
This is looking good! I don't quite understand the changes to the filtering; maybe some additional comments in `get_context_data` would be helpful? All the bits that I understand look great.  Also, I played with changing languages and setting filters in staging, and I couldn't unearth any problems.

**Comment by suecarmol** [comment]
> This is looking good! I don't quite understand the changes to the filtering; maybe some additional comments in `get_context_data` would be helpful? All the bits that I understand look great. Also, I played with changing languages and setting filters in staging, and I couldn't unearth any problems.

I had to change how the filtering worked because the partners view wasn't filtering. Hope these new comments help! Let me know in what part of the code you have further questions (if you have any).

**Comment by jsnshrmn** [comment]
Yes, the comments clear things up nicely. Thanks for that, and the tabnabbing mitigation! 🙌

**Pull request merged**